system-design/specification/applications/application-registry.md

```diff
@@ -0,0 +1,259 @@
+
+The following response example is a Margo-specific OCI image manifest following the [OCI Image Manifest Specification](https://github.com/opencontainers/image-spec/blob/v1.0.1/manifest.md) and the above defined specifics:
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.margo.app.v1+json" # this MUST be the artifactType of an OCI image manifest of a Margo Application Package,
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json", # the 'config' object MUST be empty
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2,
+    "data": "e30="
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.margo.app.description.v1+yaml", # this MUST be the artifactType of a Margo Application Description file
+      "digest": "sha256:f6b79149e6650b0064c146df7c045d157f3656b5ad1279b5ce9f4446b510bacf",
+      "size": 999,
+      "annotations": {
+        "org.opencontainers.image.title": "margo.yaml"
+      }
+    },
+    {
+      "mediaType": "text/markdown",
+      "digest": "sha256:e373b123782a2b52483a9124cc9c578e0ed0300cb4131b73b0c79612122b8361",
+      "size": 1596,
+      "annotations": {
+        "org.margo.app.resource": "metadata.catalog.application.descriptionFile", # each Margo application resource MUST be annotated with a dotted path to its definition in the Application Description file
+        "org.opencontainers.image.title": "resources/description.md"
+      }
+    },
+    {
+      "mediaType": "text/markdown",
+      "digest": "sha256:af7db4ab9030533b6cda2325247920c3659bc67a7d49f3d5098ae54a64633ec7",
+      "size": 25,
+      "annotations": {
+        "org.margo.app.resource": "metadata.catalog.application.licenseFile", # each Margo application resource MUST be annotated with a dotted path to its definition in the Application Description file
+        "org.opencontainers.image.title": "resources/license.md"
+      }
+    },
+    {
+      "mediaType": "image/jpeg",
+      "digest": "sha256:451410b6adfdce1c974da2275290d9e207911a4023fafea0e283fad0502e5e56",
+      "size": 5065,
+      "annotations": {
+        "org.margo.app.resource": "metadata.catalog.application.icon", # each Margo application resource MUST be annotated with a dotted path to its definition in the Application Description file
+        "org.opencontainers.image.title": "resources/margo.jpg"
+      }
+    },
+    {
+      "mediaType": "text/markdown",
+      "digest": "sha256:c412d143084c3b051d7ea4b166a7bfffb4550f401d89cae8898991c65e90f736",
+      "size": 42,
+      "annotations": {
+        "org.margo.app.resource": "metadata.catalog.application.releaseNotes", # each Margo application resource MUST be annotated with a dotted path to its definition in the Application Description file
+        "org.opencontainers.image.title": "resources/release-notes.md"
+      }
+    }
+  ],
+}
+```
+
+#### Margo-Specific Media Types
```

Comment:
As mentioned above, I'd expect future versions of the Margo spec to declare a couple of more Margo-specific resources and declare MediaTypes for them, replacing the use of the annotations for them.

In general filtering based on the value of a JSON document based on a fixed key is much easier and more efficient than filtering for specific keys in a key-value store and then filtering for a specific value.

To better understand it, try to filter with `jq` the above manifest to get the digest of the blob with MediaType `application/vnd.margo.app.description.v1+yaml` and then try again with `jq` to get the digest of the icon.